### PR TITLE
Add support for Snapshot Management

### LINKED ISSUE #70: [FEATURE] Support Snapshot Management
### Is your feature request related to a problem?
Need implementation of Snapshot Management API https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore/

### What solution would you like?
Support the Snapshot Management API in this provider: https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/snapshots/sm-api/#create-or-update-a-policy

### What alternatives have you considered?
Create snapshots using direct API calls or using the Dashboards UI.

### Do you have any additional context?
no

### PR DESCRIPTION
**Pull request by duboisph**
### Description

Added support for [Snapshot Management policies](https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/snapshots/sm-api/), largely based off the ISM code, but with a slightly different API.

An important difference is that a policy's Name is not the same as it's ID (`<policy_name>` + `-sm-policy` suffix), however the name is used for API requests. Within the provider code, I opted to use the policy ID for the TF state's `id`, but everything else uses the `policy_name`  (resource definition, import, api requests).

Also this API uses `POST` for create and `PUT` to update, using the same body data. Responses for `GET`, `POST` and `UPDATE` are all the same.

### Issues Resolved

Closes #70.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by duboisph** [comment]
Hi @prudhvigodithi, do you have an idea when someone could review this please?

**Comment by duboisph** [comment]
@prudhvigodithi Thank you! Meanwhile the terrafmt should be fixed.

**Comment by prudhvigodithi** [comment]
On quick thought, @duboisph do you think we should tell customer to pre-create the repo before create/update the policy ? this way its aware that this feature would tackle the SM policies, not configuring the snapshot repositories. Thank you

**Comment by duboisph** [comment]
Hi @prudhvigodithi. Indeed, good point. I've pushed an udpdate to the example. Would this be clear enough?

File: examples/resources/opensearch_sm_policy/resource.tf [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Create a snapshot repository. Make sure you also have created the bucket (eg. 
+# via `terraform-aws-modules/s3-bucket/aws`) and matching IAM role.
+resource "opensearch_snapshot_repository" "repo" {
+  name = "es-index-backups"
```
Comment (prudhvigodithi) [inline examples/resources/opensearch_sm_policy/resource.tf:4]: Please re-name to `os-index-backups`.

File: provider/resource_opensearch_sm_policy.go [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+
+	d.SetId(policyResponse.PolicyID)
+	return resourceOpensearchSMPolicyRead(d, m)
+}
+
+func resourceOpensearchSMPolicyRead(d *schema.ResourceData, m interface{}) error {
+	policyResponse, err := resourceOpensearchGetSMPolicy(d.Get("policy_name").(string), m)
+
+	if err != nil {
+		if elastic6.IsNotFound(err) || elastic7.IsNotFound(err) {
```
Comment (prudhvigodithi) [inline provider/resource_opensearch_sm_policy.go:91]: NIT: Please remove `elastic6`, AFAIK all the go files are updated to use `elastic7`. Thanks

**Comment by prudhvigodithi** [comment]
Thanks @duboisph, LGTM, approved the PR, please take care of my above 2 comments.

**Review by prudhvigodithi — APPROVED** [review]

**Comment by duboisph** [comment]
Thank you @prudhvigodithi! I've resolved your comments :)

**Comment by prudhvigodithi** [comment]
LGTM, I will merge the PR. Thanks again for your contribution @duboisph.

**Comment by msvticket** [comment]
@prudhvigodithi When will this be released?

**Comment by prudhvigodithi** [comment]
Hey @msvticket, thanks for following up I will do a quick short release today to get this feature out. 
Thanks

**Comment by prudhvigodithi** [comment]
This feature should be now in the v2.2.0 https://registry.terraform.io/providers/opensearch-project/opensearch/2.2.0, please check, thanks @duboisph for you contribution.

**Pull request merged**